public/app/ui/pages/search/SearchPage.js

```diff
@@ -12,41 +12,34 @@
             'models/ListModeSwitcher',
             'models/ListSorter',
             'models/ListPager',
-            'ui/pages/search/SearchType'
+            'util/safelyParseJson'
         ],
-        function (_, ko, qs, routes, container, DynamicRecord, ListModeSwitcher, ListSorter, ListPager, SearchType) {
+        function (_, ko, qs, routes, container, DynamicRecord, ListModeSwitcher, ListSorter, ListPager, safelyParseJson) {
```

Comment:
The major changes in this file are that a lot of it has been broken out into separate modules, and addition of the advanced mode switching mechanism.